modules/engine-prototype/src/main/scala/engine/Sequence.scala

```diff
@@ -0,0 +1,91 @@
+package engine
+
+import scala.concurrent.ExecutionContext
+
+import cats.{FlatMap, Functor}
+import cats.data.NonEmptyList
+import cats.effect.Effect
+import cats.effect.implicits._
+import cats.implicits._
+
+import fs2.Stream
+import fs2.async
+import fs2.async.mutable.Signal
+
+import monocle.{Lens, Prism}
+import monocle.macros.GenLens
+
+sealed trait Sequence
+object Sequence {
+
+  sealed trait F2 extends Sequence
+  object F2 {
+
+    final case class Ready(
+      done: List[Step.F2.Done],
+      pending: List[Step.F2.Pending],
+      current: Step.F2.Current
```

Comment:
Notice the Steps types are different for each position. `Current` will contain the Step under execution with information pertaining to the actions being taken. `Done` and `Pending` are almost the same but still distinct as so that it's not possible to mess things up.